arch/arm/neon_intrins.h

```diff
@@ -25,8 +25,8 @@
     out.val[3] = vqsubq_u16(a.val[3], b); \
 } while (0)
 
-#  if defined(__clang__) && defined(__arm__) && defined(__ANDROID__)
-/* Clang for 32-bit Android has too strict alignment requirement (:256) for x4 NEON intrinsics */
+#  if defined(__clang__) && defined(__arm__) && defined(__ANDROID__) && !defined(__GNUC__)
+/* Clang for 32-bit Android and musl-libc have too strict alignment requirement (:256) for x4 NEON intrinsics */
```

Comment:
This comment is just plain wrong... musl doesn't enforce stricter alignment, it's just that the correct header isn't yet included at this point.

Reply:
It builds with this patch. I just adjusted the comment based on the preprocessor condition on the line above. I'm open for suggestions; how would you like to phrase it?

Reply:
The `# undef` lines below are supposed to clear any preprocessor macros that toolchain headers define, Clang should always define both `__clang__` and `__GNUC__`, so the test above seems overkill like already mentioned by others.

Reply:
Oh you are right, it was meant for `!__GLIBC__` not `!__GNUC__` as  they don't define a dedicated macro to detect musl libc (their claim/reason is something like "musl libc strictly adheres to POSIX standard"). The original patch removed `__ANDROID__`, I didn't tested it after the fb commit.